src/main/java/org/sagebionetworks/bridge/services/AdminAccountService.java

```diff
@@ -0,0 +1,321 @@
+        account.setNotifyByEmail(FALSE);
+        account.setFirstName(submittedAccount.getFirstName());
+        account.setLastName(submittedAccount.getLastName());
+        account.setEmail(submittedAccount.getEmail());
+        account.setPhone(submittedAccount.getPhone());
+        account.setSynapseUserId(submittedAccount.getSynapseUserId());
+        account.setClientData(submittedAccount.getClientData());
+        account.setLanguages(submittedAccount.getLanguages());
+        account.setClientTimeZone(submittedAccount.getClientTimeZone());
+        account.setNote(submittedAccount.getNote());
+
+        RequestContext context = RequestContext.get();
+
+        // Admins and superadmins can set any organization membership
+        if (context.isInRole(Roles.ADMIN, Roles.SUPERADMIN)) {
+            account.setOrgMembership(submittedAccount.getOrgMembership());
+        }
+        // If they choose not to, or it‘s any other kind of account, set org to caller‘s organization
+        if (account.getOrgMembership() == null) {
+            account.setOrgMembership(context.getCallerOrgMembership());
+        }
+        for (String attribute : app.getUserProfileAttributes()) {
+            String value = submittedAccount.getAttributes().get(attribute);
+            account.getAttributes().put(attribute, value);
+        }
+        // Hash password if it has been supplied.
+        if (submittedAccount.getPassword() != null) {
+            try {
+                PasswordAlgorithm passwordAlgorithm = DEFAULT_PASSWORD_ALGORITHM;
+                String passwordHash = passwordAlgorithm.generateHash(submittedAccount.getPassword());
+                account.setPasswordAlgorithm(passwordAlgorithm);
+                account.setPasswordHash(passwordHash);
+                account.setPassword(null);
+            } catch (InvalidKeyException | InvalidKeySpecException | NoSuchAlgorithmException ex) {
+                throw new BridgeServiceException("Error creating password: " + ex.getMessage(), ex);
+            }
+        }
+        Set<Roles> finalRoles = updateRoles(context, submittedAccount.getRoles(), account.getRoles());
+        account.setRoles(finalRoles);
+
+        accountDao.createAccount(account);
+        
+        sendVerificationMessages(app, account);
+        return account;
+    }
+    
+    public Account updateAccount(String appId, Account account) {
+        checkNotNull(account);
+        
+        App app = appService.getApp(appId);
+        if (app == null) {
+            throw new EntityNotFoundException(App.class);
+        }
+        if (account.getId() == null) {
+            throw new EntityNotFoundException(Account.class);
+        }
+        account.setPassword(null); // don’t validate this value
+
+        AccountId accountId = AccountId.forId(appId,  account.getId());
+        Account persistedAccount = accountDao.getAccount(accountId)
+                .orElseThrow(() -> new EntityNotFoundException(Account.class));
```

Comment:
So, AccountService checks that you have rights to access this account, and we need to do that here as well.

Reply:
Added without the checks around participants and study access.